codec/src/main/java/io/netty/handler/codec/CodecOutputList.java

```diff
@@ -30,22 +33,132 @@
     private static final Recycler<CodecOutputList> RECYCLER = new Recycler<CodecOutputList>() {
         @Override
         protected CodecOutputList newObject(Handle<CodecOutputList> handle) {
-            return new CodecOutputList(handle);
+            // Size of 16 should be good enough for 99 % of all users.
+            return new CodecOutputList(handle, 16);
         }
     };
 
+    private static final Recycler.Handle<CodecOutputList> NOOP_RECYCLER  = new Recycler.Handle<CodecOutputList>() {
+        @Override
+        public void recycle(CodecOutputList object) {
+            // drop on the floor
+        }
+    };
+
+    private static final FastThreadLocal<CodecOutputListPool> DEQUE_FAST_THREAD_LOCAL =
+            new FastThreadLocal<CodecOutputListPool>() {
+                @Override
+                protected CodecOutputListPool initialValue() throws Exception {
+                    return new DefaultCodecOutputListPool(8);
+                }
+            };
+
+    private static final FastThreadLocal<CodecOutputListPool> SPECIAL_POOL_FAST_THREAD_LOCAL =
+            new FastThreadLocal<CodecOutputListPool>() {
+                @Override
+                protected CodecOutputListPool initialValue() throws Exception {
+                    return new SpecialCodecOutputListPool(16);
+                }
+            };
+
+    static CodecOutputList newLocalInstance() {
+        return DEQUE_FAST_THREAD_LOCAL.get().get();
+    }
+
+    static CodecOutputList newLocalSpecialInstance() {
+        return SPECIAL_POOL_FAST_THREAD_LOCAL.get().get();
+    }
+
+    private interface CodecOutputListPool extends Recycler.Handle<CodecOutputList> {
+        CodecOutputList get();
```

Comment:
getOrCreate? .get().get() looks a little weird.

Reply:
The whole interface will go away ... I just added it to make it easier to compare different impls for the benchmark.